addon/-private/system/snapshot.js

```diff
@@ -24,25 +24,13 @@ export default class Snapshot {
     this._hasManyIds = Object.create(null);
     this._internalModel = internalModel;
 
-    let record = internalModel.getRecord();
+    // TODO is there a way we can assign known attributes without
+    //  using `eachAttribute`? This forces us to lookup the model-class
+    //  but for findRecord / findAll these are empty and doing so at
+    //  this point in time is unnecessary.
+    internalModel.eachAttribute((keyName) => this._attributes[keyName] = internalModel.getAttributeValue(keyName));
```

Comment:
Looks like this broke `defaultValue` not being considered anymore, when the attribute hasn't been accessed yet 😔 

So given:

```js
// app/models/foo.js
export default Model.extend({
  bar: attr({ defaultValue: () => "baz" })
});
```

then

```js
let foo = store.createRecord('foo');

// this assertion breaks after upgrading to v3.2.0-beta.2 :pensive:
assert.deepEqual(foo.toJSON(), { foo: "baz" });
```

---

Currently the whole story around `defaultValue` is not ideal (e.g. #2566) and doing this in the app code by explicitly passing default values when creating a record is better, but I would still consider this a breaking change... What do you think?

Reply:
Seems good to open an issue for this, I _think_ it should be fixable...

Reply:
Should be very fixable, surprised this broke though, re-examining my changes, will open an issue.

Reply:
hrm, so I didn't pull the defaultValue logic into internalModel, nor am I sure we should. https://github.com/emberjs/data/pull/5413/files#diff-c2a90350f555423721228dfa13c137f2R130

That said, the proposal in #5419 would be a good direction to address this, as we can access each attribute via the record if it exists, and only copy lazily if it does not.